### ISSUE TITLE
Prediction of new unknowns

### ISSUE THREAD

**Issue by Tomisall** [CHENGLi-96/ChemAHNet#3]
Hello, 

Is it possible to add more details to the ReadMe to explain how to perform the prediction of new unknowns? 

I tried doing:

```python
from preprocess import process

process('example_test_real')
with open('example_test_data.pickle','rb') as file:
    from_pkl =pickle.load(file)
plk_dict = from_pkl[0]

tensor_dict = {}
for key,value in plk_dict.items():
    print(f'{key}      -----      {type(value)}')
    if type(value) == np.ndarray:
        tensor_dict[key] = torch.from_numpy(value)
    else:
        tensor_dict[key] = value

pred = model(tensor_dict)
pred_np = pred.cpu().detach().numpy()
```

And this seemed to work, but I am having trouble interpreting the output: `array([[0.48749262]], dtype=float32)`. Can you describe the process to identify the outcome for mew reactions please?

**Comment by CHENGLi-96** [comment]
Hi there,
Thanks for your interest in our work. The code snippet you provided for running the inference is correct.
Regarding the output [[0.48749262]]: this is the normalized ddG prediction. To get the actual ddG value (in kcal/mol), you need to reverse the normalization using the following formula:
actual_ddG = output * (4.5 + 4.5) - 4.5
For your specific output, the actual ddG would be approximately -0.11. Regarding the prediction of new unknown reactions: Deep learning models are highly dependent on the chemical space of their training data. If your new reactions involve different reaction centers, structural scaffolds, or mechanisms that are not covered in our training set (out-of-distribution), the direct prediction results will not be reliable. To achieve good performance on entirely new reaction classes, you would need to collect data for those specific reactions to fine-tune or retrain the model.
